__tests__/endspoints.test.js

```diff
@@ -18,4 +19,46 @@ describe('Fetching API topics requests', () => {
             })
         })
     })
-})
\ No newline at end of file
+
+    test('Status 200: GETs all EndPoints Info',()=>{
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((data)=>{
+            const endpoints = data.body
+            
+            for (let endpoint in endpoints) {
+             const endpointObject = endpoints[endpoint]
+             const length = Object.keys(endpointObject).length
+
+             switch(length) {
+                case 1:
+                  // code block
+                  expect(endpointObject).toMatchObject({
+                    description : expect.any(String),
+                })
+                  break;
+                case 2:
+                  // code block
+                  expect(endpointObject).toMatchObject({
+                    description : expect.any(String),
+                    queries : expect.any(Array),
+                })
+                  break;
+                case 3:
+                    expect(endpointObject).toMatchObject({
+                        description : expect.any(String),
+                        queries : expect.any(Array),
+                        exampleResponse : expect.any(Object),
+                    })
+                  break;
+                default:
+              }
+
```

Comment:
You could definitely make your life easier here if you require in the json file and assert that what comes back from the request looks the same. I admire this effort though!